nexus/src/app/background/init.rs

```diff
@@ -239,6 +216,37 @@ impl BackgroundTasks {
             vec![Box::new(rx_blueprint)],
         );
 
+        // Background task: inventory collector
+        //
+        // This currently depends on the "output" of the blueprint executor in
+        // order to automatically trigger inventory collection whenever the
+        // blueprint executor runs.  In the limit, this could become a problem
+        // because the blueprint executor might also depend indirectly on the
+        // inventory collector.  In that case, we may need to do something more
+        // complicated.  But for now, this works.
+        let task_inventory_collection = {
+            let collector = inventory_collection::InventoryCollector::new(
+                datastore.clone(),
+                resolver,
+                &nexus_id.to_string(),
+                config.inventory.nkeep,
+                config.inventory.disable,
+            );
+            let task = driver.register(
+                String::from("inventory_collection"),
+                String::from(
+                    "collects hardware and software inventory data from the \
+                    whole system",
+                ),
+                config.inventory.period_secs,
+                Box::new(collector),
+                opctx.child(BTreeMap::new()),
+                vec![Box::new(rx_blueprint_exec)],
```

Comment:
I think we currently have blueprint exec set to every 60s, and inventory collection set to every 600s. Is this going to 10x the inventory collection frequency?

Reply:
If a target is set, yes.  You could vie the 600s after this as "if there's nothing going on, collect at least this often".

I imagine we'll want to revisit these triggers and timeouts when we get to fully automating this but I think this is okay for now.

Reply:
Even as we're doing this by hand, we'll go from 600s to 60s as soon as we set a target, but it will never go back, right? Once we've set a target there will always be a target.

Reply:
I guess what I'm kinda wishing for here is "only trigger the collection if blueprint realization actually changed something", but that might be quite difficult.

Reply:
Agreed, but yeah, that seems hard to know.  Certainly if we get almost any errors, we have to assume something may have changed.  We could have each module in the execution part return information about whether it _might_ have done anything that would require re-inventorying.  Concretely, we'd need sled agent to tell us whether it made any changes when we did `PUT /omicron-zones`.  This seems like a potentially fine optimization, but also just an optimization (and bugs in this area would be really annoying).  I think it makes sense to wait until this becomes more of a problem.  What do you think?